### PR TITLE
Add explicit requirements for jupyter notebooks examples

### PR DESCRIPTION
**Pull request by 363734**
Add for all example notebooks the explicit list of Python packages the user should have (in addition to cpmpy) to make the notebooks run

### CODE REVIEW

**Comment by 363734** [comment]
If it is possible using the requirements file, it would be the best! It would simplify the life of the users and avoid the need to search all the commands to install all the packages separately

File: examples/nqueens_1000.ipynb [hunk 1]
```diff
@@ -12,7 +12,9 @@
     "Somebody had written on a slide that CSP solving with backtracking search, arc consistency and a good variable ordering makes solving n-queens for N=1000 'feasible'.\n",
     "\n",
     "So, this notebook checks how 'feasible' n-queens for different N is for a modern CP solver like or-tools CP-SAT.\n",
-    "(technical note: we first tuned the hyper parameters on N=100 so that the solver is performing at its best)"
+    "(technical note: we first tuned the hyper parameters on N=100 so that the solver is performing at its best)\n",
+    "\n",
+    "This example requires you to install _numpy_."
```
Comment (IgnaceBleukx) [inline examples/nqueens_1000.ipynb:17]: Numpy is a requirement of CPMpy itself, so no need to mention it here.

**Comment by Wout4** [comment]
The conclusion here was to not use the requirement file, since that would lead to many otherwise unneeded packages being installed (and we don't have time to figure out the optional requirements). So this should be good to merge now?

**Comment by IgnaceBleukx** [comment]
Btw what we can also do is to install the packages from within the notebook. 
This way you do not need to put it in the `requirements` file, but also allows you to skip manual installation.

Actuallyt, any system-command can be executed using '!' before the command.
E.g., in the first cell of the notebook we can do `! pip install plotly geopy` for example. Is this something we want?

**Comment by 363734** [comment]
I would not recommend putting the pip install within the notebook. As the user could want to define its own conda environment to test the notebook. 

It can be merge

**Comment by IgnaceBleukx** [comment]
I'm in the process of adding it to the notebooks, should be ready to merge by the end of the day

**Pull request merged**